lib/linguist/heuristics.yml

```diff
@@ -400,12 +400,12 @@ disambiguations:
     pattern: "(?i:(alter module)|(language sql)|(begin( NOT)+ atomic)|signal SQLSTATE '[0-9]+')"
   # Oracle
   - language: PLSQL
-    pattern: '(?i:\$\$PLSQL_|XMLTYPE|sysdate|systimestamp|\.nextval|connect by|AUTHID (DEFINER|CURRENT_USER)|constructor\W+function)'
+    pattern: '(?i:\$\$PLSQL_|XMLTYPE|sysdate|systimestamp|\.nextval|connect by|AUTHID (DEFINER|CURRENT_USER)|constructor\W+function|FROM DUAL)'
   # T-SQL
   - language: TSQL
     and:
     - negative_pattern: '(?i:IDENTIFIED|NUMBER|VARCHAR2|REPEAT|UNTIL|IMMEDIATE)'
-    - pattern: '(?i:(GO)|(@@)|(CREATE PROCEDURE)|BEGIN( TRY| CATCH)|OUTPUT( INSERTED)|IF|ELSE|IIF|CHOOSE|CURSOR|FETCH|DEALLOCATE|DECLARE)'
+    - pattern: '(?i:(GO)|(@@)|(CREATE PROCEDURE)|BEGIN( TRY| CATCH)|OUTPUT( INSERTED)|IF|ELSE|IIF|CHOOSE|CURSOR|FETCH|DEALLOCATE|DECLARE|ISNULL|GETDATE|SET NOCOUNT|datetime2|bit|datetimeoffset|sql_variant|uniqueidentifier|USE )'
```

Comment:
I don't think all of these are unique to TSQL. I've not tried very hard and I've already found `ISNULL` is [usable in MySQL](https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#function_isnull).

Reply:
`ISNULL` in SQL Server has 2 parameters, maybe change regex expression to handle this metric?
https://docs.microsoft.com/ru-RU/sql/t-sql/functions/isnull-transact-sql?view=aps-pdw-2016

Reply:
Detecting the number of parameters is not an easy task and the regular expression is already long. I'm not sure it would be worth it.